src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -691,7 +691,8 @@ public void ApplyLogsToDataFile(CancellationToken token, TimeSpan timeToWait)
 
                     ApplyJournalStateAfterFlush(token, lastProcessedJournal, lastFlushedTransactionId, unusedJournals, byteStringContext);
 
-                    _waj._env.SuggestSyncDataFile();
```

Comment:
@arekpalinski @ayende  This was the reason why actually the heuristic was running even if there was a write transaction in place immediately after a flush. As long as there was at least 1 full journal it would apply the Sync.